Document GitOps v1.8 RN

Srivaralakshmi opened the pull request:
[RHDEVDOCS-4961](https://issues.redhat.com/browse/RHDEVDOCS-4961): GitOps 1.8 release notes
[RHDEVDOCS-5030](https://issues.redhat.com/browse/RHDEVDOCS-5030): Update instances of kam to reflect the correct usage

- **Aligned team**: Dev Tools
- **OCP version for cherry-picking**: enterprise-`4.10` and later
- **JIRA issues**: [RHDEVDOCS-4961](https://issues.redhat.com/browse/RHDEVDOCS-4961)
- **Preview pages**: [Red Hat OpenShift GitOps release notes](https://55921--docspreview.netlify.app/openshift-enterprise/latest/cicd/gitops/gitops-release-notes.html)
- **SME Review**: Completed by @iam-veeramalla, @jaideepr97, @reginapizza, @ciiay, @harrietgrace 
- **QE review**: Completed by @varshab1210
- **Peer-review**: Completed by @rolfedh , @deerskindoll, @shipsing

Srivaralakshmi commented on modules/go-compatibility-and-support-matrix.adoc:
```diff
@@ -15,6 +15,7 @@ In the table, features are marked with the following statuses:
 |*OpenShift GitOps* 8+|*Component Versions*|*OpenShift Versions*
 
 |*Version* |*kam*    |*Helm*  |*Kustomize* |*Argo CD*|*ApplicationSet* |*Dex*     |*RH SSO* |*Notifications Controller*|
+|1.8.0    |0.0.47 TP |3.10.0 GA|4.5.7 GA   |2.6.1 GA |2.4.5 GA       |2.35.1 GA |7.5.1 GA |2.4.5 TP|4.8-4.12
```
Question to SMEs: Review the support matrix versions and their support status for 1.8.0 and provide feedback.

- Refer to the confluence [GitOps Component Matrix](https://spaces.redhat.com/display/GITOPS/GitOps+Component+Matrix) page.
- The value for the ApplicationSet column is given as **n/a**. Please confirm the correct value that must go in the OCP docs?
- Also, I do not see the *Notifications Controller* column and its values on the confluence matric page. Do we need to add this information to the confluence page as well? Confirm this value and support status as well for v.1.8.0.release.

Srivaralakshmi commented:
Hi @iam-veeramalla @jaideepr97 @varshab1210: PTAL at this PR and provide feedback. Thanks!

jaideepr97 commented on modules/gitops-release-notes-1-8-0.adoc:
```diff
@@ -0,0 +1,59 @@
+
+The current release adds the following improvements:
+
+* With this update, you can enable workload monitoring for specific Argo CD instances by setting the `.spec.monitoring.enabled` flag value to `true`. As a result, the Operator creates a `PrometheusRule` object that contains alert rules for each Argo CD component. These alert rules trigger the firing of an alert when the replica count of the corresponding component has drifted from the desired sta ...
```
Instead of `You can modify the rules as per the requirements and prevent the Operator from overwriting them.` can we just say `Changes made to the prometheusRule by the users will not be overwritten by the operator`?

varshab1210 replied:
https://issues.redhat.com/browse/GITOPS-2457 should be linked instead of https://issues.redhat.com/browse/GITOPS-2459? 
Correct me if I am wrong @jaideepr97

jaideepr97 replied:
sure
i dont know how much it matters either way

Srivaralakshmi commented:
Hi @jaideepr97: PTAL at the revised PR and provide your feedback. Thanks!

jaideepr97 commented:
LGTM after we address @varshab1210's point

Srivaralakshmi commented:
> LGTM after we address @varshab1210's point

@jaideepr97 Thanks for the LGTM. 

I have a question: I see that [GITOPS-2459](https://issues.redhat.com/browse/GITOPS-2459) is tagged with the **`release-note-item`** label and the **Release Note Text** on Jan 17, 2023. Given that, ideally, we link the issue that has these two aforementioned items with the OCP RN text items.

Request to help me understand better why we need to link this issue [GITOPS-2457](https://issues.redhat.com/browse/GITOPS-2457) instead.

jaideepr97 commented:
LGTM thanks @Srivaralakshmi

varshab1210 commented:
LGTM

rolfedh reviewed (approved):
LGTM

deerskindoll reviewed (approved):
lgtm

Srivaralakshmi commented:
@varshab1210 @jaideepr97 @iam-veeramalla @ciiay @rolfedh @deerskindoll : Requesting LGTM from you. Thanks!

Srivaralakshmi commented on modules/go-compatibility-and-support-matrix.adoc:
```diff
@@ -42,6 +43,10 @@ Some features in this release are currently in Technology Preview (TP). These ex
 |====
 |Feature |TP in OCP versions|GA in OCP versions
 
+|Multiple sources for an application
+|4.11, 4.12, 4.13
+|NA
+
```
@harrietgrace @iam-veeramalla: I have mentioned this feature in the [OCP RN for GitOps v1.8](https://55921--docspreview.netlify.app/openshift-enterprise/latest/cicd/gitops/gitops-release-notes.html#GitOps-technology-preview_gitops-release-notes.

what OCP versions must I mention in the **TP in OCP versions** column for this feature? Please check the existing data and provide feedback. Thanks!

iam-veeramalla replied:
The OCP versions should be all the supported OCP versions for v1.8 GitOps operator.

Srivaralakshmi replied:
Addressed it.

harrietgrace commented:
Looking at the future timeline for OCP support and our 1.8 release, it looks like we'll need to support OCP 4.10 on that as well. OCP 4.10 will still be supported for a month after 1.7 goes EOL, and we need to have at least one supported version of GitOps on every supported version of OCP.

I've updated the component matrix on confluence with this addition, and for your reference here's the spreadsheet where I figure all this out: https://docs.google.com/spreadsheets/d/1uHORNoJR5nbx8Oa2bwCjOkht7ZvoQ9b2F514UG_Pe5c/edit#gid=114872730

Thanks for your patience with all these changes, I really appreciate your help @Srivaralakshmi! 🙇🏻

Srivaralakshmi commented:
@iam-veeramalla @harrietgrace @varshab1210 I have added the feature names for multiple resources and Application set Progressive rollout strategy to the [OCP GitOps v1.8 RN](https://github.com/openshift/openshift-docs/pull/55921) TP table. Also added the Release note item for the Application set Progressive rollout strategy feature and included OCP 4.10 to the list of OCP versions. PTAL and provide feedback.

iam-veeramalla commented:
@Srivaralakshmi Thanks for the release notes. A couple of things more need to be added.
[GITOPS-1688](https://issues.redhat.com/browse/GITOPS-1688) - Adds support to Run GitOps operator on ARM architecture - 
[@Regina Scott](https://redhat-internal.slack.com/team/U014P8HKTMG) can you please add release notes item.
[GITOPS-2570](https://issues.redhat.com/browse/GITOPS-2570) - Please refer to the release note item section.

Srivaralakshmi commented:
> @Srivaralakshmi Thanks for the release notes. A couple of things more need to be added. [GITOPS-1688](https://issues.redhat.com/browse/GITOPS-1688) - Adds support to Run GitOps operator on ARM architecture - [@Regina Scott](https://redhat-internal.slack.com/team/U014P8HKTMG) can you please add release notes item. [GITOPS-2570](https://issues.redhat.com/browse/GITOPS-2570) - Please refer to the release note item section.

@iam-veeramalla , @reginapizza , @varshab1210 , @harrietgrace: I have added the feature name for the `must-gather` tool to the [OCP GitOps v1.8 RN](https://github.com/openshift/openshift-docs/pull/55921) TP table. I also added the Release note item for the [GITOPS-1688](https://issues.redhat.com/browse/GITOPS-1688) - Adds support to Run GitOps operator on ARM architecture and [GITOPS-2570](https://issues.redhat.com/browse/GITOPS-2570).

PTAL  at the revised PR and provide feedback.

reginapizza commented on modules/gitops-release-notes-1-8-0.adoc:
```diff
@@ -0,0 +1,76 @@
+====
+//https://github.com/argoproj/argo-cd/pull/12103
+
+* With this update, the {gitops-title} Operator and the `kam` is now available to use on ARM architecture on {product-title}. link:https://issues.redhat.com/browse/GITOPS-1688[GITOPS-1688]
```
should be the 'kam' CLI, not just 'kam'

Srivaralakshmi replied:
Addressed it.

Srivaralakshmi commented:
@reginapizza Incorporated your comments. PTAL.

shipsing reviewed (commented):
Do we need to hyphenate command-line interface? The Red Hat supplementary guide suggests the one without hyphen (command line interface).

shipsing commented on modules/gitops-release-notes-1-8-0.adoc:
```diff
@@ -0,0 +1,76 @@
+`spec.sso.provider: keycloak` is not yet supported on ARM.
+====
+
+* With this update, you can enable workload monitoring for specific Argo CD instances by setting the `.spec.monitoring.enabled` flag value to `true`. As a result, the Operator creates a `PrometheusRule` object that contains alert rules for each Argo CD component. These alert rules trigger the firing of an alert when the replica count of the corresponding component has drifted from the desired sta ...
```
This part seems a bit off `firing of an alert`. 
Suggestion:
```suggestion
* With this update, you can enable workload monitoring for specific Argo CD instances by setting the `.spec.monitoring.enabled` flag value to `true`. As a result, the Operator creates a `PrometheusRule` object that contains alert rules for each Argo CD component. These alert rules trigger an alert when the replica count of the corresponding component has drifted from the desired count for a certain amount of time. The Operator will not overwrite the changes made to the `PrometheusRule` object by the users. link:https://issues.redhat.com/browse/GITOPS-2459[GITOPS-2459]
```

Srivaralakshmi replied:
@shipsing agree with removing the word firing. But with the 2nd suggestion, it is the state and NOT the count. So no changes there.

Srivaralakshmi commented:
> command-line interface

@shipsing Yes, Refer to [IBM word usage](https://www.ibm.com/docs/en/ibm-style?topic=word-usage) and RH [SSG guideline](https://stylepedia.net/style/6.0/#c).

If you have a different reference, please share. I would like to learn more.

shipsing commented:
Thank you @Srivaralakshmi for looking into my suggestions. The PR looks good.

shipsing reviewed (approved):
/lgtm

varshab1210 commented on modules/go-compatibility-and-support-matrix.adoc:
```diff
@@ -42,16 +43,27 @@ Some features in this release are currently in Technology Preview (TP). These ex
 |NA
 
+|Argo CD Notifications controller
+|4.8, 4.9, 4.10, 4.11, 4.12
```
@harrietgrace Are we planning to support Argo CD Notifications controller and kam on OCP < 4.10?

harrietgrace replied:
The notifications controller is part of Argo CD, so it will inherit the support of the version it was introduced with. We added notifications as TP in 1.6.0 which supports 4.8 through 4.11. Once 1.6.0 goes EOL, then the notifications TP will be supported on 1.7.0 and up, which inherits 4.10 through 4.12 support. 

Does that make sense? This table might require an adjustment to show the GitOps Operator version of each feature, rather than focusing on the OCP support.

varshab1210 replied:
Okay, thanks @harrietgrace, makes sense

harrietgrace replied:
I'll make a separate jira to change this table over to GitOps versions rather than OCP versions. I don't think it's urgent enough to delay the release or rush out the change.

Srivaralakshmi replied:
@harrietgrace As per our discussion the last day, I have made the following changes in the TP table:

- Removal of kam from TP table as it is a component and not a feature and is already captured in the matrix table
- Update to TP table with GitOps versions

PTAL  at the revised PR and approve.

Cc: @varshab1210

Srivaralakshmi replied:
setting this to resolved.

iam-veeramalla commented:
/lgtm

harrietgrace commented:
Looks good! 👍

Srivaralakshmi commented:
@varshab1210 , @iam-veeramalla: Added the Known Issue. PTAL and approve.

@harrietgrace As per our discussion the last day, I have made the following changes in the TP table:

- Removal of kam from TP table as it is a component and not a feature and is already captured in the matrix table
- Update to TP table with GitOps versions

PTAL at the revised PR and approve.

shipsing reviewed (approved):
/lgtm

varshab1210 commented:
LGTM, Thanks @Srivaralakshmi

harrietgrace commented:
Nice! Looks great, thanks @Srivaralakshmi!

ciiay commented:
LGTM
Thanks @Srivaralakshmi 👏

rolfedh commented:
/cherrypick enterprise-4.10

rolfedh commented:
/cherrypick enterprise-4.11

rolfedh commented:
/cherrypick enterprise-4.12

rolfedh commented:
/cherrypick enterprise-4.13
